### ISSUE TITLE
macOS build failed

### ISSUE THREAD

**Issue by pshemass** [awakesecurity/gRPC-haskell#79]
When I try to enter nix shell with `nix-shell release.nix -A grpc-haskell.env` I'm getting error.

```
nix-shell release.nix -A grpc-haskell.env
these derivations will be built:
  /nix/store/mjgwrv707wcy25ix63lh88d70msc3lx6-remove-references-to.drv
  /nix/store/wpj053azh8hyjb7vsxbqj9l6jplmd8yy-optparse-applicative-0.14.0.0.drv
  /nix/store/81mfrc5rfnpc1y0xylpjhlrjdmp0yckw-tasty-0.11.2.5.drv
  /nix/store/qkxni0wjw9x2417m0s7jyp7aa8kjjl5x-tasty-hunit-0.9.2.drv
  /nix/store/9a3z725rng8h389xrrdjrapk1pv9hvw6-temporary-1.2.1.1.drv
  /nix/store/cqmfphcipvnplbxpkdl0xc5a1bxid1hg-mockery-0.3.5.drv
  /nix/store/pmzh3yfr5w4mlm0yxz61k1jf8l8pd77r-silently-1.2.5.drv
  /nix/store/dkn63rqlp7blsc6jik85naf3dgbil6fh-doctest-0.11.4.drv
  /nix/store/7y3vk43xdr05lh1i1bb354vr71nn3v5n-distributive-0.5.3.drv
  /nix/store/q95zaayj9fy927lmma88widjmvnkwj2k-comonad-5.0.2.drv
  /nix/store/s9insgfyac2nx5pg872bzb7kqmmwccdz-bifunctors-5.4.2.drv
  /nix/store/032jbcs4xj30sgz1503s98zlr52ywnkn-semigroupoids-5.2.1.drv
  /nix/store/0yw2xm1qwisnhk6qmpxqmf0jacqg5fhb-python2.7-Cython-0.24.1.drv
  /nix/store/d0gsxgh5c6ibfzfqwkqxvpm82akvz2i6-grpc-e2cfe9d.drv
  /nix/store/zhsps3r8134i1bsqp02m7azxi5m6bb60-python2.7-protobuf-3.2.0.drv
  /nix/store/xk1j0bj6wnbqz0jrwp7lf4q9vg09l057-python2.7-grpc-1.0.drv
  /nix/store/44m0gip2170jyy2nswdxzv890dhvrlqp-python2.7-grpc-1.0.drv
  /nix/store/0kxw473c8024gsgq1v58nf8pnzm625db-python-2.7.14-env.drv
  /nix/store/1b4b95dbzmmj2rkxrfkm4pfg5yx2r3j8-grpc-1.2.0-e2cfe9d.drv
  /nix/store/68qwbzs5c3pv0n8cxn4w5rkw2chnr9ax-profunctors-5.2.1.drv
  /nix/store/1jiqg76ag2zv36ljdayjfggjw6lyl6dk-free-4.12.4.drv
  /nix/store/2bfq66mf4vpg9bvxgzwd05yzsqv0pwwa-tasty-ant-xml-1.1.0.drv
  /nix/store/rl7lhfm8j0jgjkbz2ym1hxka9gbf78fq-tasty-smallcheck-0.8.1.drv
  /nix/store/wqfsx6fzs8f6m0arzggxf5dh8gf1cn2g-tasty-quickcheck-0.8.4.drv
  /nix/store/vfkdnlqviqa0saz64n74fq5a89p5hzv3-scientific-0.3.5.2.drv
  /nix/store/4avpf92qiil8z8j5j64di01zv6bwm91s-quickcheck-instances-0.3.12.drv
  /nix/store/dlxlm6y2vqsw68m7l7w4vnii17b7hspb-attoparsec-0.13.1.0.drv
  /nix/store/1nr1cb2701fsrf7k6sr0nbfm4m2d8zjj-parsers-0.12.7.drv
  /nix/store/yw260hsvxzfqy88c711fw8i50q2dvqrd-cryptohash-sha256-0.11.100.1.drv
  /nix/store/4lmd4by09b7s6nb260jc1qa7cx7ihhxx-hackage-security-0.5.2.2.drv
  /nix/store/5gy67dr34q94kp5q187rgpnjkrib7wpa-aeson-1.1.1.0.tar.gz.drv
  /nix/store/w1lw5cmm07lb8qgj05aj60gcbgji6asa-foldl-1.2.5.drv
  /nix/store/5im3nwha7sy4kli7riwvlb1mssn72n2q-turtle-1.3.6.drv
  /nix/store/ihffagcyjdbcp06iiirn80dfigk0i9hs-aeson-1.1.1.0-r2.cabal.drv
  /nix/store/py05cb08jbjrx3kjm46af78sg7kfy9p9-uuid-types-1.0.3.drv
  /nix/store/yvgacyg39cv2pnygig0ln06dd9pf8g0m-aeson-1.1.1.0.drv
  /nix/store/6xzclrqrdi7jknrhmknq60radc5p7l25-aeson-pretty-0.8.5.drv
  /nix/store/8lzsqdfhh26q0zw1qnpisp65l5mlr3np-proto3-wire-a938330.drv
  /nix/store/a2j9kpv1a6q2mvrz4j05s86mk62ziy6p-HTF-0.13.2.2.drv
  /nix/store/izf7qm9i5994blyf0wc0s0dlchcihlgw-adjunctions-4.3.drv
  /nix/store/bgxbsc8426smxq833aw9gjq28hq3qgd8-kan-extensions-5.0.2.drv
  /nix/store/bmc5zpl1s27s3j033phhhazvcqqbklq9-proto3-suite-c103a8c.drv
  /nix/store/d55az882hyk7klmq63fm34fw2g9hk8c1-neat-interpolation-0.3.2.1.drv
  /nix/store/dqbp6wnmhcgqv12y4ms6cnkljrqv4gka-proto3-wire-1.0.0.drv
  /nix/store/v0a8i4wga6f7qm1z56j0rlbpkl9c2kbc-lens-4.15.4.drv
  /nix/store/xgr9vqlm9zhbiswhxcpdgp1h5gvlwlvf-insert-ordered-containers-0.2.1.0.drv
  /nix/store/jpr1aac5i9hq8val2jwip5wxlxfhcmmf-swagger2-2.1.6.drv
  /nix/store/lg76hjl7bdas17h2b811014xqf1ifxc7-range-set-list-0.1.2.0.drv
  /nix/store/lwx8qvyq2d73n4x2jp25cxnw42q2jj8j-optparse-generic-1.2.1.tar.gz.drv
  /nix/store/r23mxxvjwvnhni6iqbva0zx43snxq075-optparse-generic-1.2.1.drv
  /nix/store/m0qi5dbbqp61k1f2zvhacb4v3pgjy538-proto3-suite-0.2.0.0.drv
  /nix/store/lalrg0fzy6fd50d7w8qba8y3shvma8y5-grpc-haskell-core-0.0.0.0.drv
  /nix/store/r7n7cvprcfbj3i2di9dcw5lsmlps9f4b-grpc-haskell-0.0.0.0.drv
  /nix/store/gml3l1dgy8s10hsaw0rlhjb3ygdjgrll-ghc-8.0.2-with-packages.drv
  /nix/store/ih1qxic2mibmrxj92h46b52prlzpq4bd-cabal-install-2.0.0.0.drv
  /nix/store/zyfbs2px8il9hl7bfrzqik3gzw2hlqwl-ghc-8.0.2-with-packages.drv
building '/nix/store/mjgwrv707wcy25ix63lh88d70msc3lx6-remove-references-to.drv'...
dyld: Library not loaded: /usr/lib/system/libsystem_network.dylib
  Referenced from: /nix/store/v2sbh9m4p8fa5czsw0cr1kcz8a9xx23d-Libsystem-osx-10.11.6/lib/libSystem.B.dylib
  Reason: image not found
builder for '/nix/store/mjgwrv707wcy25ix63lh88d70msc3lx6-remove-references-to.drv' failed due to signal 6 (Abort trap: 6)
cannot build derivation '/nix/store/izf7qm9i5994blyf0wc0s0dlchcihlgw-adjunctions-4.3.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/yvgacyg39cv2pnygig0ln06dd9pf8g0m-aeson-1.1.1.0.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/6xzclrqrdi7jknrhmknq60radc5p7l25-aeson-pretty-0.8.5.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/dlxlm6y2vqsw68m7l7w4vnii17b7hspb-attoparsec-0.13.1.0.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/s9insgfyac2nx5pg872bzb7kqmmwccdz-bifunctors-5.4.2.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/ih1qxic2mibmrxj92h46b52prlzpq4bd-cabal-install-2.0.0.0.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/q95zaayj9fy927lmma88widjmvnkwj2k-comonad-5.0.2.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/7y3vk43xdr05lh1i1bb354vr71nn3v5n-distributive-0.5.3.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/w1lw5cmm07lb8qgj05aj60gcbgji6asa-foldl-1.2.5.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/1jiqg76ag2zv36ljdayjfggjw6lyl6dk-free-4.12.4.drv': 1 dependencies couldn't be built
building '/nix/store/d0gsxgh5c6ibfzfqwkqxvpm82akvz2i6-grpc-e2cfe9d.drv'...
cannot build derivation '/nix/store/r7n7cvprcfbj3i2di9dcw5lsmlps9f4b-grpc-haskell-0.0.0.0.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/lalrg0fzy6fd50d7w8qba8y3shvma8y5-grpc-haskell-core-0.0.0.0.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/xgr9vqlm9zhbiswhxcpdgp1h5gvlwlvf-insert-ordered-containers-0.2.1.0.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/bgxbsc8426smxq833aw9gjq28hq3qgd8-kan-extensions-5.0.2.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/v0a8i4wga6f7qm1z56j0rlbpkl9c2kbc-lens-4.15.4.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/d55az882hyk7klmq63fm34fw2g9hk8c1-neat-interpolation-0.3.2.1.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/wpj053azh8hyjb7vsxbqj9l6jplmd8yy-optparse-applicative-0.14.0.0.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/r23mxxvjwvnhni6iqbva0zx43snxq075-optparse-generic-1.2.1.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/1nr1cb2701fsrf7k6sr0nbfm4m2d8zjj-parsers-0.12.7.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/68qwbzs5c3pv0n8cxn4w5rkw2chnr9ax-profunctors-5.2.1.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/m0qi5dbbqp61k1f2zvhacb4v3pgjy538-proto3-suite-0.2.0.0.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/dqbp6wnmhcgqv12y4ms6cnkljrqv4gka-proto3-wire-1.0.0.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/lg76hjl7bdas17h2b811014xqf1ifxc7-range-set-list-0.1.2.0.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/vfkdnlqviqa0saz64n74fq5a89p5hzv3-scientific-0.3.5.2.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/032jbcs4xj30sgz1503s98zlr52ywnkn-semigroupoids-5.2.1.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/jpr1aac5i9hq8val2jwip5wxlxfhcmmf-swagger2-2.1.6.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/81mfrc5rfnpc1y0xylpjhlrjdmp0yckw-tasty-0.11.2.5.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/qkxni0wjw9x2417m0s7jyp7aa8kjjl5x-tasty-hunit-0.9.2.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/wqfsx6fzs8f6m0arzggxf5dh8gf1cn2g-tasty-quickcheck-0.8.4.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/9a3z725rng8h389xrrdjrapk1pv9hvw6-temporary-1.2.1.1.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/5im3nwha7sy4kli7riwvlb1mssn72n2q-turtle-1.3.6.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/py05cb08jbjrx3kjm46af78sg7kfy9p9-uuid-types-1.0.3.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/gml3l1dgy8s10hsaw0rlhjb3ygdjgrll-ghc-8.0.2-with-packages.drv': 1 dependencies couldn't be built
error: build of '/nix/store/0kxw473c8024gsgq1v58nf8pnzm625db-python-2.7.14-env.drv', '/nix/store/1b4b95dbzmmj2rkxrfkm4pfg5yx2r3j8-grpc-1.2.0-e2cfe9d.drv', '/nix/store/gml3l1dgy8s10hsaw0rlhjb3ygdjgrll-ghc-8.0.2-with-packages.drv', '/nix/store/zyfbs2px8il9hl7bfrzqik3gzw2hlqwl-ghc-8.0.2-with-packages.drv' failed
```

I'm pretty new to nix and haskell too but I assume that this is something with `dyld: Library not loaded: /usr/lib/system/libsystem_network.dylib` so I digged in and found this https://github.com/NixOS/nix/issues/2244 based on that I know that it's fixed but since in the project you pinned nixpkg I'm not getting the fix. Is that correct?
I was trying to change `rev` and `sha256` in `./nixpkg.nix` but no change in behavior. I tried even put there some bogus data but it doesn't seem to have any effect.

Could someone help me with that?

Additional info:
```
nix-env --version
nix-env (Nix) 2.2.2

nix-channel --list
nixpkgs https://nixos.org/channels/nixpkgs-unstable
```
MacOS 10.14.4

**Comment by Gabriel439** [comment]
Yeah, this is due to us using an older version of `nixpkgs` (which we specify [here](https://github.com/awakesecurity/gRPC-haskell/blob/master/nixpkgs.nix)).  Upgrading `nixpkgs` would fix the problem might require some work to update `gRPC-haskell` to build against the latest `nixpkgs`

**Comment by pshemass** [comment]
that was my assumption that changing rev and sha5 in `./nixpkgs` will let me fixed that (or move to different problem) but when I did following steps I still see this problem and I don't understand why.

```
git clone https://github.com/awakesecurity/gRPC-haskell.git

cd gRPC-haskell

nix-prefetch-git https://github.com/nixos/nixpkgs-channels.git refs/heads/nixos-19.03
{
  "url": "https://github.com/nixos/nixpkgs-channels.git",
  "rev": "c21f08bfedde0f7d00a5ebc85455fb28b9037932",
  "date": "2019-05-15T20:01:03-04:00",
  "sha256": "0sndhal0a61zzgc8k2lhba356mkrgq0hwf5aa61f6kvpvn4isjmg",
  "fetchSubmodules": false
}

 nix-prefetch-url "https://github.com/NixOS/nixpkgs/archive/c21f08bfedde0f7d00a5ebc85455fb28b9037932.tar.gz"
1f7g0qx2samf8cg91hjf9015h31vfkq390gnw775z37x4hlc4pzq
```

then I change the ./nixpkgs to be 

```
import ./fetch-nixpkgs.nix {
  rev          = "c21f08bfedde0f7d00a5ebc85455fb28b9037932";
  sha256       = "1f7g0qx2samf8cg91hjf9015h31vfkq390gnw775z37x4hlc4pzq";
  outputSha256 = "13ydgpzl5nix4gc358iy9zjd5nrrpbpwpxmfhis4aai2zmkja3ak";
}
```

and when I run `nix-shell release.nix -A grpc-haskell.env` i still getting the same error. 

what I'm doing wrong?

**Comment by pshemass** [comment]
I figured out that sha256 field it's ignored and only outputSha256 is taken into account in https://github.com/awakesecurity/gRPC-haskell/blob/master/fetch-nixpkgs.nix#L23 so changed outputSha256 to something incorrect and then did rerun and I got message about incorrect hash 

```
nix-shell release.nix -A grpc-haskell.env
error: hash mismatch in file downloaded from 'https://github.com/NixOS/nixpkgs/archive/c21f08bfedde0f7d00a5ebc85455fb28b9037932.tar.gz':
  wanted: sha256:1f7g0qx2samf8cg91hjf9015h31vfkq390gnw775z37x4hlc4pzq
  got:    sha256:0sndhal0a61zzgc8k2lhba356mkrgq0hwf5aa61f6kvpvn4isjmg
(use '--show-trace' to show detailed location information)
```

then I changed that `nixpkgs.nix` to be 
```
import ./fetch-nixpkgs.nix {
  rev          = "c21f08bfedde0f7d00a5ebc85455fb28b9037932";
  sha256       = "1f7g0qx2samf8cg91hjf9015h31vfkq390gnw775z37x4hlc4pzq";
  outputSha256 = "0sndhal0a61zzgc8k2lhba356mkrgq0hwf5aa61f6kvpvn4isjmg";
}
``` 

and now I'm geting errors about libs which make sense to
```
error: Package ‘range-set-list-0.1.3’ in /nix/store/zqgw6afzz4v8jac4gl9agnw8hkvgazbi-source/pkgs/development/haskell-modules/hackage-packages.nix:184454 is marked as broken, refusing to evaluate.
```

I will try to migrate to new version.

**Comment by Gabriel439** [comment]
@pshemass: It might be worth updating `nixpkgs.nix` to match the latest version of https://nixos.wiki/wiki/How_to_fetch_Nixpkgs_with_an_empty_NIX_PATH, which simplifies it to only require a `sha256` (which is what used to be the `outputSha256`)

**Comment by pshemass** [comment]
I took revision for `nixpkgs` from your other project `proto3-suite` and after couple of tries with dependencies I came up with PR https://github.com/awakesecurity/gRPC-haskell/pull/80 let me know how bad that is.

l manage to build and run examples successfully  but tests are failing.

**Comment by cocreature** [comment]
I believe this might be fixed by #84

**Comment by cocreature** [comment]
Closing since I think this is fixed by updating our nixpkgs. Feel free to reopen if you still encounter this on master.
